mitxgraders/helpers/mitmath/calc.py

```diff
@@ -307,6 +302,10 @@ def evaluator(formula,
     >>> evaluator("inf", variables={'inf': float('inf')}, allow_inf=True)[0]
     inf
     """
+    variables = {} if variables is None else variables
+    functions = {} if functions is None else functions
+    suffixes = {} if suffixes is None else suffixes
```

Comment:
> So, I set these to the defaults to simplify both testing and use of evaluator by authors. 
> Now to get `pi`, `e` and `sin`, we need to do a bunch more typing.

I'm ok changing this back, but some things to consider:
- (Opinion): it's nice to keep things isolated / self-contained.
- In practice, none of the graders are affected by this change because they all use local variables / functions / suffixes. 

  There are a few isolated tests of `evaluator` by itself, and this change did break one of them: that test relied on `factorial(500.50` to return `inf`. But that test was a bit hard to understand unless you happen to know that `factorial(500.5)` is inf, so I used this as an opportunity to refactor that.

**But:** If you find yourself using `evaluator` in isolation for random development / testing purposes, I can see keeping the defaults and am happy to put them back.

Reply:
I agree that in practice, this changes nothing in the implementation, and with your opinion that things should be self-contained. However, we are exposing evaluator to problem authors if they desire to use it for anything, and I do routinely find myself using evaluator in isolation for dev/testing ;-)